castai/resource_node_template.go

```diff
@@ -54,6 +55,7 @@ const (
 	FieldNodeTemplateSpotInterruptionPredictionsEnabled       = "spot_interruption_predictions_enabled"
 	FieldNodeTemplateSpotInterruptionPredictionsType          = "spot_interruption_predictions_type"
 	FieldNodeTemplateStorageOptimized                         = "storage_optimized"
+	FieldNodeTemplateStorageOptimizedEnum                     = "storage_optimized_enum"
```

Comment:
Suggest naming `FieldNodeTemplateStorageOptimizedStatus` as enum suffix doesn't add value and later on would be confusing